pkg/reboot/calculator.go

```diff
@@ -106,14 +116,83 @@ func (s *safeTimeCalculator) calcMinTimeAssumeRebooted() error {
 	s.log.Info("calculated minTimeToAssumeNodeRebooted is:", "minTimeToAssumeNodeRebooted", minTime)
 	s.minTimeToAssumeNodeRebooted = minTime
 
-	if s.timeToAssumeNodeRebooted < minTime {
-		err := fmt.Errorf("snr agent can't start: the requested value for SafeTimeToAssumeNodeRebootedSeconds is too low")
-		s.log.Error(err, err.Error(), "requested SafeTimeToAssumeNodeRebootedSeconds", s.timeToAssumeNodeRebooted, "minimal calculated value for SafeTimeToAssumeNodeRebootedSeconds", minTime)
+	//update related logic of min time on configuration if necessary
+	if err := s.manageSafeRebootTimeInConfiguration(ctx, minTime); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// manageSafeRebootTimeInConfiguration does two things:
+//  1. It sets Status.MinSafeTimeToAssumeNodeRebootedSeconds in case it's changed by latest calculation.
+//  2. It Adds/removes config.Status.SpecSafeTimeOverriddenWarning if necessary.
+func (s *safeTimeCalculator) manageSafeRebootTimeInConfiguration(ctx context.Context, minTime time.Duration) error {
+	minTimeSec := int(minTime.Seconds())
+	config, err := s.getConfig()
+	if err != nil {
 		return err
 	}
+	orgConfig := config.DeepCopy()
+	prevMinRebootTimeSec := config.Status.MinSafeTimeToAssumeNodeRebootedSeconds
+	if prevMinRebootTimeSec != minTimeSec {
+		config.Status.MinSafeTimeToAssumeNodeRebootedSeconds = minTimeSec
+	}
+	//Manage condition
+	meta.SetStatusCondition(&config.Status.Conditions, metav1.Condition{
+		Type:   v1alpha1.SafeTimeToAssumeNodeRebootedOverriddenConditionType,
+		Status: metav1.ConditionFalse,
+		Reason: "SafeTimeToAssumeNodeRebootedSeconds is valid or empty",
```

Comment:
` Reason`  should be a short camel case programmatic identifier. ` Message`  is for human readable longer explanation.

Reply:
would it be cleaner to have this in an else block of the if below?

Reply:
>would it be cleaner to have this in an else block of the if below?

It's fine by me, IIRC usually logically unneeded else blocks are frowned upon in code reviews so I try to avoid them mostly for reviews sake.

Reply:
Fair enough, probably this is an edge case. Setting a default value first, and then conditionally modifying it is fine. Not sure though if we have a "default" here... 🤔  Also, I guess I had logging and eventing for the "default" in mind, then it would be wrong for sure.